### DISCUSSION TITLE
Is there a documentation about the `dir` syntax for tasks?

### DISCUSSION THREAD

**Discussion by rik** [jdx/mise#1723] [Q&A]
I've only found [one reference](https://mise.jdx.dev/tasks/toml-tasks.html) to this option. Because it uses `{{cwd}}`, it's indicating there's a specific syntax that could be used but I haven't found it.

Is that the case?

**Comment by Marukome0743** [comment]
The `dir` property is a Tera-rendered configuration value that controls the working directory used to execute the task.

By default, a task runs from the config root—the directory containing the `mise.toml` that defines it. To run from the directory where the user invoked mise, use:

```toml
[tasks.test]
run = "pwd"
dir = "{{cwd}}"
```

`MISE_ORIGINAL_CWD` is also passed to the task as an environment variable when the original invocation directory is needed inside the script.

The current documentation now covers this under “Specifying which directory to use” in the TOML tasks documentation.

I verified the behavior against the current main branch by invoking two tasks from a nested subdirectory:

- A task without `dir` ran from the directory containing `mise.toml`.
- A task with `dir = "{{cwd}}"` ran from the nested directory where mise was invoked.

So `{{cwd}}` specifically means the original invocation directory; it is not required for the normal project-root behavior.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
